argo/providers/vai/provider.go

```diff
@@ -249,60 +219,126 @@ func (vaip VAIProvider) DeleteRun(_ context.Context, _ VAIProviderConfig, _ stri
 	return nil
 }
 
+func buildPipelineJob(ctx context.Context, providerConfig VAIProviderConfig, runScheduleDefinition RunScheduleDefinition, mutatePipelineJob func(context.Context, VAIProviderConfig, *aiplatformpb.PipelineJob) error) (*aiplatformpb.PipelineJob, error) {
+	parameters := make(map[string]*aiplatformpb.Value, len(runScheduleDefinition.RuntimeParameters))
+	for name, value := range runScheduleDefinition.RuntimeParameters {
+		parameters[name] = &aiplatformpb.Value{
+			Value: &aiplatformpb.Value_StringValue{
+				StringValue: value,
+			},
+		}
+	}
```

Comment:
I think we can use https://github.com/sky-uk/kfp-operator/blob/master/apis/pipelines/utils.go#L159 here

```suggestion
        parameters := pipelines.ToMap(runScheduleDefinition.RuntimeParameters, func(rt RuntimeParameter) (string, *aiplatformpb.Value) {
            return rt.name, &aiplatformpb.Value{
			Value: &aiplatformpb.Value_StringValue{
				StringValue: value,
			}
        })
```

Reply:
As `runScheduleDefinition.RuntimeParameters` is a Map I have introduced a MapValues function for this